api/drivers/qemu/driver.js

```
@@ -166,20 +178,42 @@ class QemuDriver extends Driver {
    * @return {Promise[Image]} resolves to the new default image
    */
   createImage(imageToCreate) {
-    return imageToCreate;
-  }
-
-  /**
-   * Retrieve the machine's data
-   *
-   * @method refresh
-   * @param {machine} Machine model
-   * @return {Promise[Machine]}
-   */
-  refresh(machine) {
-    return new Promise((resolve) => {
-      machine.status = 'running';
-      return resolve(machine);
+    return new Promise((resolve, reject) => {
+      return Promise.props({
+        config: ConfigService.get('qemuServiceURL', 'qemuServicePort', 'qemuPath'),
+        image: MachineService.getDefaultImage()
+      })
+        .then((res) => {
+          return request({
+            url: 'http://' + res.config.qemuServiceURL + ':' + res.config.qemuServicePort + '/images',
+            json: true,
+            method: 'POST',
+            body: {
+              buildFrom: imageToCreate.buildFrom,
+              name: res.config.qemuPath + imageToCreate.name
+            }
+          });
+        })
+        .then((res) => {
+
+          if (!res.name && !res.buildFrom) {
+            return reject('Something goes wrong on qemu-manager');
```

In case of an error, qemu-manager should return an error code. This will make the code fall into the nearest catch.
In general, we don't want to handle errors in `then`.